IBX-5113 - Anonymous access to store (#1919)
ezsystems/developer-documentation: docs/permissions/permission_use_cases.md

To control which commerce functionalities are available to store users, you can grant 
or prevent them access to individual components.

Out of the box, [[= product_name_com =]] comes with the *Storefront User* Role that is assigned to anonymous users and grants them the following permissions:

> - `product/view`, `product_type/view` and `catalog/view`, to allow them viewing a product 

Reviewer: ```suggestion
- `product/view`, `product_type/view` and `catalog/view`, to allow them to view a product 
```

list and product details

> - `cart/view`, `cart/create` and `cart/edit` with the `CartOwner` Limitation set to `self`, to allow them adding items to their own shopping cart, modifying their cart, and deleting it

Reviewer: ```suggestion
- `cart/view`, `cart/create` and `cart/edit` with the `CartOwner` Limitation set to `self`, to allow them to add items to their own shopping cart, modify their cart, and delete it
```

- `checkout/view`, `checkout/create`, `checkout/update` and `checkout/delete`, to allow 

> them proceeding to checkout and interacting with the checkout process

Reviewer: ```suggestion
them to proceed to checkout and interact with the checkout process
```

> You can modify the default Roles by preventing anonymous users from being able to proceed with the checkout process, and creating the *Registered Buyer* Role that enables logged-in users to purchase products. 

Reviewer: ```suggestion
You can modify the default Roles by preventing anonymous users from being able to proceed with the checkout process,
and creating the *Registered Buyer* Role that enables logged-in users to purchase products. 
```

> You could do this by moving permissions that relate to checkout from the *Storefront User* Role to the *Registered Buyer* Role, as well granting the latter with the `user/register` and `user/login` permissions which control access to registration and login.

Reviewer: ```suggestion
You could do this by moving permissions that relate to checkout from the *Storefront User* Role to the *Registered Buyer* Role,
and granting *Registered Buyer* with the `user/register` and `user/login` permissions which control access to registration and login.
```
